netZooPy/lioness/io.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import print_function
+import math
+from random import sample
+import time
+import pandas as pd
+from scipy.stats import zscore
+from .timer import Timer
+import numpy as np
+from netZooPy.panda import calculations as calc
+import sys
+
+def check_expression_integrity(df):
+    """Check data integrity
+    - Number of NA
+
+    Args:
+        df (dataframe): gene expression dataframe
+    """
+
+    # check that for each
+    if (df.isna().sum(axis = 1)>(len(df.columns)-3)).any():
+        sys.exit('Too many nan in gene expression (need more than 1 sample to compute coexpression)')
+
+def read_ppi(ppi_fn, tf_list = None):
```

Comment:
@violafanfani quick question before you merge. Are you defining these generic functions for LIONESS and then using them in OTTER?

Reply:
yes, in theory these functions could be used for PANDA or any other method too. I am trying to separate the classes from the io functions

Reply:
ok, would't then be possible to use the functions from PANDA by calling 'from PANDA import read_ppi' in OTTER?

Reply:
I had been smart, and I had forgotten

Reply:
oh, no, but I am now realizing that I am still not using the panda io functions

Reply:
ok then just pay attention to these IO functions because the ones in PANDA are quite developed as they include a number of edges cases. It is very easy to make a mistake here (TF/Gene dataframe alignment, table conversions etc).

Reply:
yes, but the issue is that the actual PANDA functions are still accessible just by instantiating a PANDA. In the test (test_otter lines 67-75) I am checking that the data obtained by our IOs is the same as if we were using the PANDA ones. Before being able to use the same functions, we would need to separate the IOs from the PANDA class. I will add it as an issue/enhancement

Reply:
I will add a warning in readme, and will go ahead with the merging. 
Does that sound good?

Reply:
yep that makes sense 👍